lib/gtm.js

```diff
@@ -0,0 +1,29 @@
+import { useEffect } from 'react';
+
+/**
+ * Google tag manager related utility functions
+ */
+
+/**
+ * @param  {...any} args - data to feed to dataLayer.push
+ */
+export function pushToDataLayer(...args) {
+  if (typeof window === 'undefined') return; // Skip in SSR
+
+  if (!window.dataLayer) window.dataLayer = [];
+  window.dataLayer.push(...args);
+}
+
+/**
+ * @param {object} props - data to send to pushDataLayer on component mount
+ */
+export function PushToDataLayer({ ...props }) {
+  useEffect(() => {
+    // Make sure all rendered elements has been flushed to DOM
+    setTimeout(() => {
+      pushToDataLayer(props);
```

Comment:
hmm...do we really need to wait for a macro task?
That means when component did mount, it doesn't mean that dom updated.

Reply:
Good point. I added this because by the time this is invoked, the title is not flushed to DOM yet (probably because <Head> also works asynchronously). But I agree it does not make sense here.

Maybe the developer should handle timeout in Google Tag Manager instead?

Reply:
For example, `ReactGA` pass all parameter by manually passing props rather than part from page. That's might be a more react way. 

I'm okay with both ways but noted that what we waited is head meta elements

Reply:
Yeah I agree. But the point of using Google Tag Manager instead of triggering `ga()` in source code is to enable designers and developers to add whatever tracking and track whatever event they want without deploying new versions of source code. That's why I would like to focus on the timing of 'trigger' (which can be integrated to any 'tag') rather than how to get Google analytics event done.